src/box/memtx_hash.cc

```diff
@@ -285,12 +285,25 @@ memtx_hash_index_random(struct index *base, uint32_t rnd, struct tuple **result)
 {
 	struct memtx_hash_index *index = (struct memtx_hash_index *)base;
 	struct light_index_core *hash_table = &index->hash_table;
-
-	*result = NULL;
-	uint32_t k = light_index_random(hash_table, rnd);
-	if (k == light_index_end)
+	struct txn *txn = in_txn();
+	struct space *space = space_by_id(base->def->space_id);
+	if (memtx_hash_index_size(base) == 0) {
+		*result = NULL;
+		memtx_tx_track_full_scan(txn, space, base);
 		return 0;
-	*result = light_index_get(hash_table, k);
+	}
+
+	do {
+		uint32_t k = light_index_random(hash_table, rnd);
```

Comment:
You must increment `rnd` each time, otherwise you'll get the same `k` each time.

Reply:
Yeah, missed that, my test was broken and it didn't catch this out, sorry.